4.5 release page (#1138)
godotengine/godot-website: collections/_release_4_5/entry-systems-xr-application-spacewarp-reporting-for-duty.md

---
type: entry
section: systems
subsection: XR
rank: 4
importance: 4
anchor: application-spacewarp-reporting-for-duty
title: Application SpaceWarp, reporting for duty
text: |

>   Mobile headset resources can be sparse, so it’s important to know how to optimize the compute budget. Because XR requires a high refresh rate, developers are left with a limited timeframe to render. But what we could use frame synthesis to our advantage? We could render at half-rate while the GPU synthesizes the next one, keeping at whole the targeted framerate while giving ourselves effectively much more time.

Reviewer: ```suggestion
  Mobile headset resources can be sparse, so it’s important to know how to optimize the compute budget. Because XR requires a high refresh rate, developers are left with a limited timeframe to render. But what if we could use frame synthesis to our advantage? We could render at half-rate while the GPU synthesizes the next frame, keeping the targeted framerate as a whole while giving ourselves effectively much more time.
```

Reviewer 2: Yeah, I think "the next one" should be "the next frame".

Reviewer: Indeed there's nothing being referred back to, updating

>   Mobile headset resources can be sparse, so it’s important to know how to optimize the compute budget. Because XR requires a high refresh rate, developers are left with a limited timeframe to render. But what if we could use frame synthesis to our advantage? We could render at half-rate while the GPU synthesizes the next one, keeping the targeted framerate as a whole while giving ourselves effectively much more time.

Reviewer: ```suggestion
  Mobile headset resources can be sparse, so it’s important to know how to optimize the compute budget. Because XR requires a high refresh rate, developers are left with a limited timeframe to render. But what if we could use frame synthesis to our advantage? We could render at half-rate while the GPU synthesizes the next frame, keeping the targeted framerate as a whole while giving ourselves effectively much more time.
```

>   As mobile headset resources can be sparse, it is important to know how to optimize the compute budget. Considering XR requires a high refresh rate for a multitude of reasons, this leaves developper with a limited timeframe to render. But what if it could be possible to use frame synthesis to our advantage? We could render at half-rate, giving ourselves way more time to process, while leveraging the GPU to generate the next one.

Reviewer 3: ```suggestion
  Mobile headset resources can be sparse, so it's important to know how to optimize the compute budget. Because XR requires a high refresh rate, developers are left with a limited timeframe to render. But what we could use frame synthesis to our advantage? We could render at half-rate, giving ourselves way more time to process, while the GPU generates the next one.
```
Fixed a few grammar and spelling things, as well as sentence flow.

It feels a bit ambiguous to me what "the next one" is referring to; I presume it means "the next frame"? With this paragraph, I'm also finding myself confused by the discussion of refresh rate as a whole. The beginning suggests that a high refresh rate is essential, but the last sentence seems to suggest that rendering at half the rate is somehow an option.

Author: I rephrased the sentence, but applied your changes. Thanks!

  That’s exactly what Meta’s [Application SpaceWarp](https://developers.meta.com/horizon/blog/introducing-application-spacewarp/) does. We now support this technology with the latest release of our [OpenXR vendors plugin](https://godotengine.org/article/godot-openxr-vendors-plugin-400/), thanks to the implementation of motion vectors in the Mobile renderer.

>   As OpenXR just released the multi-vendor [Frame Synthesis extension](https://registry.khronos.org/OpenXR/specs/1.1/html/xrspec.html#XR_EXT_frame_synthesis), we expect support for non Pico and Quest headsets in the future.

Reviewer 4: I don't feel like the "for non Pico and Quest headsets" makes it clear which headsets currently support this. How about we flip it around like:

```suggestion
  That’s exactly what Meta’s [Application SpaceWarp](https://developers.meta.com/horizon/blog/introducing-application-spacewarp/) does on Meta and Pico headsets. We now support this technology with the latest release of our [OpenXR vendors plugin](https://godotengine.org/article/godot-openxr-vendors-plugin-400/), thanks to the implementation of motion vectors in the Mobile renderer.

  As OpenXR just released the multi-vendor [Frame Synthesis extension](https://registry.khronos.org/OpenXR/specs/1.1/html/xrspec.html#XR_EXT_frame_synthesis), we expect support for more headsets in the future.
```
